dexbot/strategies/ataxia.py

```diff
@@ -0,0 +1,277 @@
+            self['dynamic_upper_bound'] = upper_bound
+            self['dynamic_size'] = size
+            self['last_bounds_check'] = time.time()
+        else:
+            if time.time() - (self['last_bounds_check'] or 0) > BOUNDS_CHECK_TIME:
+                self['last_bounds_check'] = time.time()
+                # recompute bounds and size
+                new_upper_bound, new_lower_bound = self.compute_bounds()
+                new_ladder = self.create_ladder(new_upper_bound, new_lower_bound)
+                new_size = self.compute_size(new_ladder)
+                # but don't bother redoing the orders unless bounds/size have actually moved significantly
+                if (abs(new_upper_bound-(self['dynamic_upper_bound'] or 0))/new_upper_bound > WIGGLE/100.0
+                    or abs(new_lower_bound-(self['dynamic_lower_bound'] or 0))/new_lower_bound > WIGGLE/100.0
+                        or abs(new_size-(self['dynamic_size'] or 0))/new_size > WIGGLE/100.0):
+                    self.log.info('dynamic parameters changed, purging orderbook')
+                    self.cancel_all()
+                    lower_bound = new_lower_bound
+                    self['dynamic_lower_bound'] = lower_bound
+                    upper_bound = new_upper_bound
+                    self['dynamic_upper_bound'] = upper_bound
+                    size = new_size
+                    self['dynamic_size'] = size
+                    ladder = new_ladder
+                else:
+                    # change too small: stick to old values
+                    lower_bound = self['dynamic_lower_bound']
+                    upper_bound = self['dynamic_upper_bound']
+                    size = self['dynamic_size']
+                    ladder = self.create_ladder(upper_bound, lower_bound)
+            else:
+                # its not time to recheck, stick to old values
+                lower_bound = self['dynamic_lower_bound']
+                upper_bound = self['dynamic_upper_bound']
+                size = self['dynamic_size']
+                ladder = self.create_ladder(upper_bound, lower_bound)
+        self.last_check = time.time()
+        # prepare up and down ladders
+        downladder = ladder
+        upladder = ladder.copy()
+        upladder.reverse()
+        new_order = True
+        total_orders = 0
+        # now do the orders
+        while new_order:
+            new_order = False
+            self.account.refresh()
+            highest_buy, lowest_sell = self.spread_zone(upper_bound, lower_bound)
+            self.log.debug("highest_buy = {} lowest_sell = {}".format(highest_buy, lowest_sell))
+            # do max one order on each side, then cycle outer loop (i.e. check back
+            # with market whether things have shifted)
+            for price, ladder_size in downladder:
+                if price > lowest_sell:
+                    if self.check_at_price(1/price):  # sell orders are inverted
```

Comment:
Logic error here. Sell orders are inverted but in check_at_price() the sell orders are also inverted, meaning you also need to do the price inversion in check_at_price() also.

Reply:
No, they are inverted in self.orders (a wrapper for account.openorders) so I have to invert so they match (ie. we are comparing inverted to inverted, which is what we want). I don't have to invert elsewhere as self.market_sell does it for me.
Trust me, it works ;-)

Reply:
I tested it and it gave me an infinite loop because check_at_price was constantly returning True even if there was an order. The problem lies with the inversion